rdt/transformers/base.py

```diff
@@ -210,10 +213,11 @@ def _build_output_columns(self, data):
         self.output_columns = self.get_output_columns()
 
         # make sure none of the generated `output_columns` exists in the data
-        data_columns = set(data.columns)
-        while data_columns & set(self.output_columns):
+        output_columns = set(self.output_columns) - set(self.columns)
+        while set(output_columns) & set(data.columns):
             self.column_prefix += '#'
             self.output_columns = self.get_output_columns()
+            output_columns = set(self.output_columns) - set(self.columns)
```

Comment:
should this be `self.output_columns`?

Reply:
no, this variable control the while loop